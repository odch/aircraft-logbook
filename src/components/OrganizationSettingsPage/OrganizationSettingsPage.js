import React from 'react'
import PropTypes from 'prop-types'
import Header from '../../containers/HeaderContainer'
import OrganizationSettings from '../../containers/OrganizationSettingsContainer'

const OrganizationSettingsPage = ({ match }) => (
  <React.Fragment>
    <Header />
    <OrganizationSettings match={match} />
  </React.Fragment>
)

OrganizationSettingsPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.object.isRequired
  }).isRequired
}

export default OrganizationSettingsPage
