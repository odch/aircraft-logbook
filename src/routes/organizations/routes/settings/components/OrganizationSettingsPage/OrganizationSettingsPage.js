import React from 'react'
import PropTypes from 'prop-types'
import Header from '../../../../../../containers/HeaderContainer'
import OrganizationSettings from '../../containers/OrganizationSettingsContainer'

const OrganizationSettingsPage = ({ router: { match } }) => (
  <React.Fragment>
    <Header />
    <OrganizationSettings match={match} />
  </React.Fragment>
)

OrganizationSettingsPage.propTypes = {
  router: PropTypes.shape({
    match: PropTypes.shape({
      params: PropTypes.object.isRequired
    }).isRequired
  })
}

export default OrganizationSettingsPage
