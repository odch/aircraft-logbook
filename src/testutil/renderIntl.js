import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
