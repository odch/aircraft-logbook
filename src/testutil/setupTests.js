import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import moment from 'moment'
import momentTimezone from 'moment-timezone'
import { init as initFeatureToggles } from '../util/featureToggles'

enzyme.configure({ adapter: new Adapter() })

moment.locale('de')
momentTimezone.locale('de')

initFeatureToggles()
