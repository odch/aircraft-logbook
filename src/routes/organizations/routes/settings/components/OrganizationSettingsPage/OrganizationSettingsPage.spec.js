import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import { renderIntlMaterial } from '../../../../../../testutil/renderIntl'
import OrganizationSettingsPage from './OrganizationSettingsPage'
import { MuiPickersUtilsProvider } from '@material-ui/pickers'
import MomentUtils from '@date-io/moment'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('settings', () => {
        describe('components', () => {
          describe('OrganizationSettingsPage', () => {
            it('renders correctly', () => {
              const state = {
                firebase: {
                  auth: {
                    isEmpty: false,
                    email: 'test@opendigital.ch'
                  },
                  profile: {}
                },
                firestore: {
                  ordered: {
                    organizationMembers: [
                      {
                        firstname: 'Max',
                        lastname: 'Muster',
                        id: 'user1'
                      },
                      {
                        firstname: 'Hans',
                        lastname: 'Keller',
                        id: 'user2'
                      }
                    ]
                  }
                },
                main: {
                  app: {
                    organizations: [{ id: 'my_org' }]
                  }
                },
                organizationSettings: {
                  createMemberDialog: {
                    open: false
                  },
                  deleteMemberDialog: {
                    open: false
                  },
                  exportFlightsForm: {
                    submitting: false,
                    data: {
                      startDate: '2019-08-01',
                      endDate: '2019-08-31'
                    }
                  },
                  members: {
                    page: 0
                  }
                }
              }

              const store = configureStore()(state)

              const props = {
                router: {
                  match: {
                    params: {
                      organizationId: 'my_org'
                    }
                  }
                }
              }

              const renderedValue = renderIntlMaterial(
                <MuiPickersUtilsProvider utils={MomentUtils}>
                  <Provider store={store}>
                    <Router>
                      <OrganizationSettingsPage {...props} />
                    </Router>
                  </Provider>
                </MuiPickersUtilsProvider>,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
