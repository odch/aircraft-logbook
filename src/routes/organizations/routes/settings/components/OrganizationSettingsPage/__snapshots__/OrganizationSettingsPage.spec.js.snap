// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes settings components OrganizationSettingsPage renders correctly 1`] = `
"<header class=\\"MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary\\">
  <div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\">
    <h6 class=\\"MuiTypography-root Header-grow-2 MuiTypography-h6 MuiTypography-colorInherit\\"><a class=\\"Header-homeLink-4\\" href=\\"/\\">Open Digital Flugreisebuch</a></h6>
    <div><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" data-cy=\\"user-button\\" aria-haspopup=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
  </div>
</header>
<div class=\\"injectIntl(OrganizationSettings)-container-99\\">
  <h4 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom\\" data-cy=\\"organization-title\\">my_org</h4><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Mitglied hinzufügen</span><span class=\\"MuiTouchRipple-root\\"></span></button>
  <ul class=\\"MuiList-root MuiList-padding\\">
    <li class=\\"MuiListItem-container\\">
      <div class=\\"MuiListItem-root MuiListItem-secondaryAction\\">
        <div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">Muster Max</span>
          <p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary\\"></p>
        </div>
      </div>
      <div class=\\"MuiListItemSecondaryAction-root\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
    </li>
    <li class=\\"MuiListItem-container\\">
      <div class=\\"MuiListItem-root MuiListItem-secondaryAction\\">
        <div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">Keller Hans</span>
          <p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary\\"></p>
        </div>
      </div>
      <div class=\\"MuiListItemSecondaryAction-root\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
    </li>
  </ul>
  <div class=\\"MuiTablePagination-root\\">
    <div class=\\"MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters\\">
      <div class=\\"MuiTablePagination-spacer\\"></div><span class=\\"MuiTypography-root MuiTablePagination-caption MuiTypography-caption MuiTypography-colorInherit\\">1-2 of 2</span>
      <div class=\\"MuiTablePagination-actions\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z\\"></path></svg></span></button></div>
    </div>
  </div>
  <form>
    <div>
      <div class=\\"MuiFormControl-root MuiTextField-root ExportFlightsForm-datePicker-151 MuiFormControl-marginNormal\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">Startdatum</label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd\\" type=\\"text\\" value=\\"01.08.2019\\">
          <div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
        </div>
      </div>
      <div class=\\"MuiFormControl-root MuiTextField-root ExportFlightsForm-datePicker-151 MuiFormControl-marginNormal\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">Enddatum</label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd\\" type=\\"text\\" value=\\"31.08.2019\\">
          <div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
        </div>
      </div>
    </div>
    <div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Flüge exportieren</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
  </form>
  <hr class=\\"MuiDivider-root injectIntl(OrganizationSettings)-divider-100\\">
  <div class=\\"injectIntl(OrganizationSettings)-deleteButtonContainer-101\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" data-cy=\\"organization-delete-button\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg>Organisation löschen</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
</div>"
`;
